crates/xilem_web/src/class.rs

```diff
@@ -0,0 +1,118 @@
+use std::{borrow::Cow, marker::PhantomData};
+
+use xilem_core::{Id, MessageResult};
+
+use crate::{
+    interfaces::{sealed::Sealed, Element},
+    ChangeFlags, Cx, View, ViewMarker,
+};
+
+/// A trait to make the class adding functions generic over collection type
+pub trait IntoClasses {
+    fn into_classes(self, classes: &mut Vec<Cow<'static, str>>);
+}
+
+impl IntoClasses for String {
+    fn into_classes(self, classes: &mut Vec<Cow<'static, str>>) {
+        classes.push(self.into());
+    }
+}
+
+impl IntoClasses for &'static str {
+    fn into_classes(self, classes: &mut Vec<Cow<'static, str>>) {
+        classes.push(self.into());
+    }
+}
```

Comment:
```suggestion
}

impl IntoClasses for Cow<'static, str> {
    fn into_classes(self, classes: &mut Vec<Cow<'static, str>>) {
        classes.push(self);
    }
}
```

(Simply because it's a low-hanging fruit)